src/vs/workbench/parts/extensions/electron-browser/extensionsActions.ts

```diff
@@ -1857,6 +1889,111 @@ export class ConfigureWorkspaceFolderRecommendedExtensionsAction extends Abstrac
 	}
 }
 
+export class AddToWorkspaceRecommendationsAction extends AbstractConfigureRecommendedExtensionsAction {
+
+	private static readonly AddLabel = localize('addToWorkspaceRecommendations', "Add to workspace recommendations");
+	private static readonly AddingLabel = localize('addingToWorkspaceRecommendations', "Adding to workspace recommendations...");
+	private static readonly AddedLabel = localize('addedToWorkspaceRecommendations', "Added to workspace recommendations");
+
+	private static readonly AddClass = 'extension-action add-to-workspace-recommendations';
+	private static readonly AddingClass = 'extension-action add-to-workspace-recommendations adding';
+	private static readonly AddedClass = 'extension-action add-to-workspace-recommendations added';
+
+	private _extension: IExtension;
+	get extension(): IExtension { return this._extension; }
+	set extension(extension: IExtension) { this._extension = extension; this.update(); }
+
+	private disposables: IDisposable[] = [];
+
+	constructor(
+		@IFileService fileService: IFileService,
+		@IWorkspaceContextService contextService: IWorkspaceContextService,
+		@IEditorService editorService: IEditorService,
+		@IJSONEditingService jsonEditingService: IJSONEditingService,
+		@ITextModelService textModelResolverService: ITextModelService,
+		@ICommandService private commandService: ICommandService,
+		@INotificationService private notificationService: INotificationService,
+	) {
+		super(
+			'extensions.add-to-workspace-recommendations',
+			AddToWorkspaceRecommendationsAction.AddLabel,
+			contextService,
+			fileService,
+			editorService,
+			jsonEditingService,
+			textModelResolverService
+		);
+		this.class = AddToWorkspaceRecommendationsAction.AddClass;
+		this.enabled = false;
+		this.disposables.push(this.contextService.onDidChangeWorkbenchState(() => this.update()));
```

Comment:
Take the case of an extension for which this action is enabled.
User adds a folder to current workspace and this workspace already recommends the extension.
Now when user runs this action the `pickFolder` promise will return all workspace folders

I wonder if we should instead listen to the `onDidChangeWorkspaceFolders` event.

We will have access to whether a workspace was added, removed or changed.
Changed => we don't need to react
Removed => we should update if the current state of the action is disabled
Added => we should update if the current state of the action is enabled